src/auth_service/config.py

```diff
@@ -15,113 +15,123 @@
-        oidc_authority_url + "/OIDC/userinfo"  # type: ignore
-    )
-    oidc_client_id: str = "ghga-data-portal"
-
-    # the URL used as source for internal claims
-    organization_url: HttpUrl = "https://ghga.de"  # type: ignore
-
-    db_url: SecretStr = "mongodb://mongodb:27017"  # type: ignore
-    db_name: str = "user-management"
-    users_collection: str = "users"
-    claims_collection: str = "claims"
-
-    service_instance_id: str = "1"
-    kafka_servers: list[str] = ["kafka:9092"]
-
-    # the topic and type of the event announcing dataset deletions
-    dataset_deletion_event_topic: str = "metadata_datasets"
-    dataset_deletion_event_type: str = "dataset_deleted"
+    service_name: str = Field(
+        default=SERVICE_NAME, description="Short name of this service"
+    )
+
+    run_auth_adapter: bool = Field(default=False, description="Run as auth adapter")
+
+    auth_key: Optional[str] = Field(
+        default=None,
+        description="internal public key for user management"
+        " (key pair for auth adapter)",
+    )
+
+    api_ext_path: str = Field(
+        default="/api/auth",
+        description="external API path for the user management"
+        " as seen by the auth adapter",
+    )
+
+    auth_ext_keys: Optional[str] = Field(
+        default=None,
+        description="external public key set for auth adapter"
+        " (not used for user management)",
+    )
+    auth_ext_algs: list[str] = Field(
+        default=["RS256", "ES256"],
+        description="allowed algorithms for signing external tokens",
+    )
+
+    basic_auth_credentials: Optional[str] = Field(
+        default=None,
+        description="credentials for basic authentication, separated by whitespace",
+    )
+    basic_auth_realm: str = Field(
+        default="GHGA Data Portal", description="realm for basic authentication"
+    )
+
+    allow_read_paths: list[str] = Field(
+        default=["/.well-known/*", "/service-logo.png"],
+        description="paths that are public or use their own authentication mechanism",
+    )
+    allow_write_paths: list[str] = Field(
+        default=[],
+        description="paths for writing that use their own authentication mechanism",
+    )
+
+    include_apis: list[Literal["users", "claims"]] = Field(
+        default=["users"],
+        description="If not run as auth adapter, which APIs should be provided."
+        " If no APIs are specified, run the event consumer.",
+    )
+
+    add_as_data_stewards: list[Union[str, dict]] = Field(
+        default=[],
+        description="a list of external IDs of data stewards or user objects"
+        " to seed the claims repository with",
+    )
+
+    oidc_authority_url: HttpUrl = Field(
+        default="https://proxy.aai.lifescience-ri.eu",
+        description="external OIDC authority URL used by the auth adapter",
+    )
+    oidc_userinfo_endpoint: Optional[HttpUrl] = Field(
+        default="https://proxy.aai.lifescience-ri.eu/OIDC/userinfo",
```

Comment:
This was only the relative endpoint path without the base URL before, is this intended?

Reply:
> This was only the relative endpoint path without the base URL before, is this intended?

No, it was also absolute before. It just reused the class scoped attribute to define the default value which cannot be done. But it's exactly the same default value.

Needed to change it because in a function call you can't use class scoped attribute names. But I think that's also better because it looked like it was calculated dynamically, which was not the case.

Reply:
Thanks for the explanation, sometimes there's a bit of a limited context when looking at diffs.
Should be good to go.